cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -392,10 +390,82 @@ def _validate_feature_id(self, feature_id: str, df_name: str):
 
         return
 
+    def _get_matrix_format(self, matrix: Union[np.ndarray, sparse.spmatrix]) -> str:
+        """
+        Given a matrix, returns the format as one of: csc, csr, coo, dense
+        or unknown.
+
+        This mimics the scipy.sparse `format` property, but extends it to
+        support ndarray and other classes AnnData may proxy the matrix with.
+        """
+
+        # Note: the AnnData proxy classes DO support the `format_str` property, but
+        # doing a slice seemed safer, if less performant.  Using `format_str`, which
+        # currently works, uses private API:
+        #
+        # >>> return getattr(matrix, "format_str", "dense)
+        #
+        format = "unknown"
+        if self.adata.n_obs == 0 or self.adata.n_vars == 0:
```

Comment:
Why would this imply that the matrix is dense?

Reply:
it implies it doesn't matter, so I randomly choose dense.  I don't think it matters what is returned....

Reply:
Ah ok, I get it now. I believe such case should be rejected by the validator already, but since the check might happen later in the code, it makes sense 👍

Reply:
From my reading of the code, the validator does not check elsewhere for an empty matrix (X or raw.X).  Caveat: I did a quick skim of the code.